Sources/LiveViewNative/Property Wrappers/LiveBinding.swift

```diff
@@ -0,0 +1,215 @@
+    /// See ``LiveBinding`` for a discussion of how the underlying binding name is determined and an example of this usage.
+    public init(attribute: AttributeName) {
+        self.bindingNameSource = .attribute(attribute)
+    }
+    
+    /// Creates a `LiveBinding` by decoding its name from a container.
+    ///
+    /// This initializer should be used when the live binding is used as part of a view modifier.
+    /// See ``LiveBinding`` for an example of how this is used.
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let name = try container.decode(String.self)
+        self.bindingNameSource = .fixed(name)
+    }
+    
+    /// The value of the binding.
+    ///
+    /// Setting this property will send an event to the server to update its value as well.
+    public var wrappedValue: Value {
+        get {
+            return data.getValue(from: liveViewModel, bindingName: bindingName)
+        }
+        nonmutating set {
+            // update the local value
+            data.mode = .local(newValue)
+            // update the view model, which will send an update to the backend
+            let encoder = FragmentEncoder()
+            // todo: if encoding fails, what should happen?
+            try! newValue.encode(to: encoder)
+            liveViewModel.setBinding(bindingName, to: encoder.unwrap() as Any)
+        }
+    }
+    
+    /// A SwiftUI `Binding` that provides read and write access to the underlying data.
+    ///
+    /// Access this binding with the dollar sign prefix. If a property is declared as `@LiveBinding(name: "value") var value`, then the projected value binding can be access as `$value`.
+    public var projectedValue: Binding<Value> {
+        Binding {
+            return self.wrappedValue
+        } set: { newValue in
+            self.wrappedValue = newValue
+        }
+    }
+}
+
+extension LiveBinding: DynamicProperty {
+    public func update() {
+        // don't need to do anything, just need to conform to DynamicProperty to make sure our @StateObject gets installed
+    }
+}
+
+extension LiveBinding {
+    enum BindingNameSource {
+        case fixed(String)
+        case attribute(AttributeName)
+    }
+}
+
+extension LiveBinding {
+    class Data: ObservableObject {
+        var mode: Mode = .uninitialized
+        private var cancellable: AnyCancellable?
+        
+        func getValue(from liveViewModel: LiveViewModel, bindingName: String) -> Value {
+            switch mode {
+            case .uninitialized:
+                cancellable = liveViewModel.bindingUpdatedByServer
+                    .filter { $0.0 == bindingName }
+                    .sink { [unowned self] _ in
+                        self.mode = .needsUpdateFromViewModel
+                        self.objectWillChange.send()
+                    }
+                fallthrough
```

Comment:
It may be beneficial to refactor this to avoid `fallthrough`. I personally think it makes this harder to read and could lead to issues if someone inadvertently reorders the cases.